faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java

```diff
@@ -331,18 +384,14 @@ private Duration tickToTime(int tick) {
   }
 
   @SneakyThrows
-  public ReplayData parse() {
-    byte[] data = readReplayData(path);
-    if (data == null) {
-      // TODO the call above should maybe throw an exception
-      throw new IllegalStateException("Could not read replay data");
-    }
+  private void parse() {
+    readReplayData(path);
     try (LittleEndianDataInputStream dataStream = new LittleEndianDataInputStream(new ByteArrayInputStream(data))) {
       parseHeader(dataStream);
       parseTicks(dataStream);
+    } catch (Exception e) {
```

Comment:
Is this IOException or do we have som others occuring? Also I'd throw and remove the Sneakythrows

Reply:
I am not sure about throwing because then it would fail just for replays with some weird mod that the parser can't handle but the game could when played.

Reply:
Whether the game can be played or not should not be relevant for a library which task it is to parse replays.
Imagine you create the parser. It fails to parse, but your code doesn't know because you just log it. What now? You are operating on an invalid object, which might cause other issues such as NullPointerException when you try to access the results.

Reply:
Hmm okay then I will probably break out the parsing of the headers and the parsing of the body as the header parsing of both the scfa and the faf header are needed to actually play the replay and are the main components used by the client and this reduces the code duplication.
The body parsing is not needed for then playing the replay so it wouldn't be good to have it fail because it couldn't parse the body which isn't needed.